rust/lance/src/io/local.rs

```diff
@@ -110,20 +110,8 @@ impl ObjectReader for LocalObjectReader {
             // Safety: `buf` is set with appropriate capacity above. It is
             // written to below and we check all data is initialized at that point.
             unsafe { buf.set_len(range.len()) };
-            #[cfg(unix)]
-            let bytes_read = file.read_at(buf.as_mut(), range.start as u64)?;
-            #[cfg(windows)]
-            let bytes_read = file.seek_read(buf.as_mut(), range.start as u64)?;
-            if bytes_read != range.len() {
-                return Err(Error::IO {
-                    message: format!(
-                        "failed to read all bytes from file: expected {}, got {}",
-                        range.len(),
-                        bytes_read
-                    ),
-                    location: location!(),
-                });
-            }
+            file.read_exact_at(buf.as_mut(), range.start as u64)?;
```

Comment:
This isn't the cause of the bug, but I did realize we are gambling a bit by using `read_at`. From the [`read_at` docs](https://doc.rust-lang.org/std/os/unix/fs/trait.FileExt.html#tymethod.read_at):

> Note that similar to [File::read](https://doc.rust-lang.org/std/fs/struct.File.html#method.read), it is not an error to return with a short read.

Instead of just erroring when the OS doesn't fill the buffer, we can just use a method that will handle this for us.